pkg/gofr/logging/logger_test.go

```diff
@@ -104,29 +106,60 @@ func TestLogger_LevelWarn(t *testing.T) {
 	infoLog := testutil.StdoutOutputForFunc(printLog)
 	errLog := testutil.StderrOutputForFunc(printLog)
 
-	if strings.ContainsAny(infoLog, "NOTICE|INFO|DEBUG") && !strings.Contains(errLog, "ERROR") {
-		// Warn Log Level will not contain  DEBUG,INFO, NOTICE logs
-		t.Errorf("TestLogger_LevelDebug Failed!")
+	levels := []Level{DEBUG, INFO, NOTICE}
+
+	for i, l := range levels {
+		assert.Equal(t, false, strings.Contains(infoLog, l.String()), "TEST[%d], Failed.\n%s", i,
+			fmt.Sprintf("unexpected %s log", l))
 	}
 
 	assertMessageInJSONLog(t, errLog, "Test Error Log")
 }
 
 func TestLogger_LevelFatal(t *testing.T) {
-	printLog := func() {
+	// running the failing part only when a specific env variable is set
+	if os.Getenv("EXITER") == "1" {
 		logger := NewLogger(FATAL)
+
 		logger.Debugf("%s", "Test Debug Log")
 		logger.Infof("%s", "Test Info Log")
+		logger.Logf("%s", "Test Log")
 		logger.Noticef("%s", "Test Notice Log")
 		logger.Warnf("%s", "Test Warn Log")
 		logger.Errorf("%s", "Test Error Log")
+		logger.Fatalf("%s", "Test Fatal Log")
+
+		return
 	}
 
-	infoLog := testutil.StdoutOutputForFunc(printLog)
-	errLog := testutil.StderrOutputForFunc(printLog)
+	//nolint:gosec // starting the actual test in a different subprocess
+	cmd := exec.Command(os.Args[0], "-test.run=TestLogger_LevelFatal")
```

Comment:
This ia dirty hack… but why not

Reply:
I would have preferred something like this

https://github.com/gofr-dev/gofr/issues/844#issuecomment-2230145018

Reply:
correct me if I am wrong.. but what I understood from your point is: you want `exitor` as a field in `logger` struct, and in `Fatal` methods, we can provide value for it as `true`, and in `logf` method we will check if `exitor` is true, then os.Exit(). Right?

If yes, I believe that will also not resolve our problem of testing, and we would need to write tests the same way as we are doing now (using subprocess).
Or, are you suggesting that instead of using NewLogger in test, we would create a logger struct manually?

Reply:
Let me write something. I'll be back soon to explain